lib/reek/smells/attribute.rb

```diff
@@ -1,4 +1,6 @@
+require_relative 'smell_configuration'
```

Comment:
I think I’m ok with using constants from `SmellDetector` once `smell_detector.rb` is required, as long as `SmellDetector` is not a namespacing-only `Module`. I think my reasoning is as follows:

* if `SmellDetector` is a class, then
  * non-module/class constants in that class can be expected to be available once `smell_detector.rb` is required;
  * modules/classes embedded inside that class are for private use by that class,
  * modules/classes that are supposed to be used from outside of `SmellDetector` should live in their own files (like `smell_detector/foo.rb`) and be required explicity;
* if `Reek` is a top namespacing module, then either
  * (more realistically) `require 'reek'` should load enough of `reek` to expose whatever we consider to be our public API, or
  * (more of a pipe-dream) all of our public API should be available through methods defined on `Reek` and `require 'reek'` should not be expected to load any of the underlying classes (even though it in practice would to let those methods work).